var/spack/repos/builtin/packages/py-jax/package.py

```diff
@@ -23,23 +23,53 @@ class PyJax(PythonPackage):
+    version(
+        "0.3.23",
+        sha256="bff436e15552a82c0ebdef32737043b799e1e10124423c57a6ae6118c3a7b6cd",
+        deprecated=True,
+    )
+    version(
+        "0.2.25",
+        sha256="822e8d1e06257eaa0fdc4c0a0686c4556e9f33647fa2a766755f984786ae7446",
+        deprecated=True,
+    )
 
+    depends_on("python@3.9:", when="@0.4.14:", type=("build", "run"))
     depends_on("python@3.8:", when="@0.4:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
+    depends_on("py-ml-dtypes@0.2:", when="@0.4.14:", type=("build", "run"))
+    depends_on("py-ml-dtypes@0.1:", when="@0.4.9:", type=("build", "run"))
+    depends_on("py-ml-dtypes@0.0.3:", when="@0.4.7:", type=("build", "run"))
+    depends_on("py-numpy@1.22:", when="@0.4.14:", type=("build", "run"))
+    depends_on("py-numpy@1.21:", when="@0.4.7:", type=("build", "run"))
     depends_on("py-numpy@1.20:", when="@0.3:", type=("build", "run"))
     depends_on("py-numpy@1.18:", type=("build", "run"))
     depends_on("py-opt-einsum", type=("build", "run"))
+    depends_on("py-scipy@1.9:", when="@0.4.19:", type=("build", "run"))
+    depends_on("py-scipy@1.7:", when="@0.4.7:", type=("build", "run"))
     depends_on("py-scipy@1.5:", when="@0.3:", type=("build", "run"))
     depends_on("py-scipy@1.2.1:", type=("build", "run"))
+    depends_on("py-importlib-metadata@4.6:", when="@0.4.11: ^python@:3.9", type=("build", "run"))
 
     # See _minimum_jaxlib_version in jax/version.py
-    jax_to_jaxlib = {"0.4.3": "0.4.2", "0.3.23": "0.3.15", "0.2.25": "0.1.69"}
-
-    for jax, jaxlib in jax_to_jaxlib.items():
-        depends_on(f"py-jaxlib@{jaxlib}:", when=f"@{jax}", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.19:", when="@0.4.21:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.14:", when="@0.4.15:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.11:", when="@0.4.12:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.7:", when="@0.4.8:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.6:", when="@0.4.7:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.4:", when="@0.4.5:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.2:", when="@0.4.3:", type=("build", "run"))
+    depends_on("py-jaxlib@0.4.1:", when="@0.4.2:", type=("build", "run"))
+    depends_on("py-jaxlib@0.3.22:", when="@0.3.24:", type=("build", "run"))
+    depends_on("py-jaxlib@0.3.15:", when="@0.3.18:", type=("build", "run"))
+    depends_on("py-jaxlib@0.3.14:", when="@0.3.15:", type=("build", "run"))
+    depends_on("py-jaxlib@0.3.7:", when="@0.3.8:", type=("build", "run"))
+    depends_on("py-jaxlib@0.3.2:", when="@0.3.7:", type=("build", "run"))
+    depends_on("py-jaxlib@0.3.0:", when="@0.3.2:", type=("build", "run"))
+    depends_on("py-jaxlib@0.1.74:", when="@0.2.26:", type=("build", "run"))
+    depends_on("py-jaxlib@0.1.69:", when="@0.2.18:", type=("build", "run"))
```

Comment:
This might be controversial. We're going from single version pinning to lower bound only. This means that the oldest jax could try to build with the newest jaxlib. I haven't confirmed whether or not that is possible.